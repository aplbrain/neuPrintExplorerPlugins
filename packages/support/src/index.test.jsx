import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { sortRois, metaInfoError, TablePaginationActions } from '.';

const ROIList = [
  '123',
  'foo',
  'foo',
  'bar',
  '123',
  '12bar',
  'foo'
];
const ROIList2 = [
  'foo',
  'bar',
  '123',
  '12bar'
]

Enzyme.configure({adapter: new Adapter()});

function doNothing() {}

test('ROIs correctly sorted by sortRois', () => {
  expect(ROIList.sort(sortRois)).toEqual(["bar", "foo", "foo", "foo", "123", "123", "12bar"]);
  expect(ROIList2.sort(sortRois)).toEqual(["bar", "foo", "123", "12bar"]);
});

const errorMessage = 'the wheels fell off';
test('metaInfoError returns correct redux object', () => {
  expect(metaInfoError(errorMessage)).toEqual({
    type: 'META_INFO_ERROR',
    error: errorMessage
  });
});

test('TablePaginationActions shows next link disabled', () => {
  const raw = <TablePaginationActions count={50} onPageChange={doNothing} rowsPerPage={5} page={10} />;
  const component = renderer.create(raw);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const wrapper = mount(raw);
  expect(wrapper.find('button').at(0).props().disabled).toBe(false);
  expect(wrapper.find('button').at(3).props().disabled).toBe(true);

});


