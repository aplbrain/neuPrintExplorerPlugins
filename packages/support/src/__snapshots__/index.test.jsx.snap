// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBox uses correct color 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "color": "#000",
      "display": "inline-flex",
      "flexDirection": "column",
      "height": "20px",
      "justifyContent": "center",
      "margin": "0px",
      "minHeight": "20px",
      "minWidth": "10px",
      "overflow": "visible",
      "width": "10px",
    }
  }
  title="100%"
>
  alt text
</div>
`;

exports[`TablePaginationActions shows next link disabled 1`] = `
<div
  className="TablePaginationActions-root-1"
>
  <button
    aria-label="First Page"
    className="MuiButtonBase-root-8 MuiIconButton-root-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiIconButton-label-7"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-11"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
        />
        <path
          d="M24 24H0V0h24v24z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-20"
    />
  </button>
  <button
    aria-label="Previous Page"
    className="MuiButtonBase-root-8 MuiIconButton-root-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiIconButton-label-7"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-11"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-20"
    />
  </button>
  <button
    aria-label="Next Page"
    className="MuiButtonBase-root-8 MuiButtonBase-disabled-9 MuiIconButton-root-2 MuiIconButton-disabled-6"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="-1"
    type="button"
  >
    <span
      className="MuiIconButton-label-7"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-11"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
    </span>
  </button>
  <button
    aria-label="Last Page"
    className="MuiButtonBase-root-8 MuiButtonBase-disabled-9 MuiIconButton-root-2 MuiIconButton-disabled-6"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="-1"
    type="button"
  >
    <span
      className="MuiIconButton-label-7"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-11"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
    </span>
  </button>
</div>
`;
